test/mlir/onnx/onnx_decompose_convtranspose.mlir

```diff
@@ -0,0 +1,207 @@
+    // CHECK:           [[VAR_15_:%.+]] = "onnx.Pad"([[VAR_9_]]#1, [[VAR_13_]], [[VAR_14_]]) {mode = "constant"} : (tensor<1x1x1x3xf32>, tensor<8xi64>, tensor<f32>) -> tensor<1x1x3x3xf32>
+    // CHECK-DAG:       [[VAR_16_:%.+]] = "onnx.Concat"([[VAR_12_]], [[VAR_15_]], [[VAR_9_]]#2) {axis = 2 : si64} : (tensor<1x1x3x3xf32>, tensor<1x1x3x3xf32>, tensor<1x1x1x3xf32>) -> tensor<1x1x7x3xf32>
+    // CHECK-DAG:       [[VAR_17_:%.+]] = onnx.Constant dense<1> : tensor<3xi64>
+    // CHECK-NOT: separator of consecutive DAGs
+    // CHECK-DAG:       [[VAR_18_:%.+]]:3 = "onnx.Split"([[VAR_16_]], [[VAR_17_]]) {axis = 3 : si64} : (tensor<1x1x7x3xf32>, tensor<3xi64>) -> (tensor<1x1x7x1xf32>, tensor<1x1x7x1xf32>, tensor<1x1x7x1xf32>)
+    // CHECK-DAG:       [[VAR_19_:%.+]] = onnx.Constant dense<[0, 0, 0, 0, 0, 0, 0, 1]> : tensor<8xi64>
+    // CHECK-DAG:       [[VAR_20_:%.+]] = onnx.Constant dense<0.000000e+00> : tensor<f32>
+    // CHECK-NOT: separator of consecutive DAGs
+    // CHECK-DAG:       [[VAR_21_:%.+]] = "onnx.Pad"([[VAR_18_]]#0, [[VAR_19_]], [[VAR_20_]]) {mode = "constant"} : (tensor<1x1x7x1xf32>, tensor<8xi64>, tensor<f32>) -> tensor<1x1x7x2xf32>
+    // CHECK-DAG:       [[VAR_22_:%.+]] = onnx.Constant dense<[0, 0, 0, 0, 0, 0, 0, 1]> : tensor<8xi64>
+    // CHECK-DAG:       [[VAR_23_:%.+]] = onnx.Constant dense<0.000000e+00> : tensor<f32>
+    // CHECK:           [[VAR_24_:%.+]] = "onnx.Pad"([[VAR_18_]]#1, [[VAR_22_]], [[VAR_23_]]) {mode = "constant"} : (tensor<1x1x7x1xf32>, tensor<8xi64>, tensor<f32>) -> tensor<1x1x7x2xf32>
+    // CHECK:           [[VAR_25_:%.+]] = "onnx.Concat"([[VAR_21_]], [[VAR_24_]], [[VAR_18_]]#2) {axis = 3 : si64} : (tensor<1x1x7x2xf32>, tensor<1x1x7x2xf32>, tensor<1x1x7x1xf32>) -> tensor<1x1x7x5xf32>
+    // CHECK-DAG:       [[VAR_26_:%.+]] = "onnx.Conv"([[VAR_25_]], [[VAR_7_]], [[VAR_0_]]) {auto_pad = "NOTSET", dilations = [1, 1], group = 1 : si64, kernel_shape = [3, 3], pads = [2, 2, 2, 2], strides = [1, 1]} : (tensor<1x1x7x5xf32>, tensor<2x1x3x3xf32>, none) -> tensor<*xf32>
+    // CHECK-DAG:       [[VAR_27_:%.+]] = onnx.Constant dense<[0, 0, 0, 0, 0, 0, 1, 0]> : tensor<8xi64>
+    // CHECK-DAG:       [[VAR_28_:%.+]] = onnx.Constant dense<0.000000e+00> : tensor<f32>
+    // CHECK-NOT: separator of consecutive DAGs
+    // CHECK-DAG:       [[VAR_29_:%.+]] = "onnx.Pad"([[VAR_26_]], [[VAR_27_]], [[VAR_28_]]) {mode = "constant"} : (tensor<*xf32>, tensor<8xi64>, tensor<f32>) -> tensor<1x2x10x7xf32>
+    // CHECK-DAG:       [[VAR_30_:%.+]] = onnx.Constant dense<[0, 0, 0, 0, 0, 0, 0, 1]> : tensor<8xi64>
+    // CHECK-DAG:       [[VAR_31_:%.+]] = onnx.Constant dense<0.000000e+00> : tensor<f32>
+    // CHECK:           [[VAR_32_:%.+]] = "onnx.Pad"([[VAR_29_]], [[VAR_30_]], [[VAR_31_]]) {mode = "constant"} : (tensor<1x2x10x7xf32>, tensor<8xi64>, tensor<f32>) -> tensor<1x2x10x8xf32>
+    // CHECK:           return [[VAR_32_]] : tensor<1x2x10x8xf32>
+  }
+
```

Comment:
@imaihal please make sure to have a test with unknown dimensions to make sure the compilation does not fail in such a situation. Thanks!

Reply:
Done. However, currently spatial dimensions need to be static. One of the reason for this is an issue in shape inference. I want to solve this following PR.  Thanks.